ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/SourceBuilderTest.java

```diff
@@ -668,6 +689,51 @@ public void shouldBuildTableWithCorrectStoreName() {
     verify(materializationFactory).create(keySerde, valueSerde, "base-Reduce");
   }
 
+  @Test
+  public void shouldRegisterSourceSchemaOnChangelogTopicWhenChangelogIsNotForced() throws IOException, RestClientException {
+    // Given:
+    final ParsedSchema mockSchema = mock(ParsedSchema.class);
+    final String sourceSubject = TOPIC_NAME + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX;
+    when(queryBuilder.getKsqlConfig())
+        .thenReturn(KSQL_CONFIG.cloneWithPropertyOverwrite(
+            ImmutableMap.of(KsqlConfig.SCHEMA_REGISTRY_URL_PROPERTY, "localhost:1234")));
+    when(queryBuilder.getQueryId()).thenReturn(new QueryId("ID"));
+    when(srClient.getAllVersions(sourceSubject)).thenReturn(ImmutableList.of(1));
+    when(srClient.getByVersion(sourceSubject, 1, false))
+        .thenReturn(new io.confluent.kafka.schemaregistry.client.rest.entities.Schema(sourceSubject, 1, 1, null, null, "schema"));
+    when(srClient.getSchemaById(1)).thenReturn(mockSchema);
+    when(srClient.getAllSubjects()).thenReturn(ImmutableSet.of(sourceSubject));
+    givenUnwindowedSourceTable(false);
+
+    // When:
+    tableSource.build(planBuilder);
+
+    // Then:
+    verify(srClient).register("_confluent-ksql-default_query_ID-base-Reduce-changelog-value", mockSchema);
+  }
+
+  @Test
+  public void shouldNotRegisterSourceSchemaOnChangelogTopicWhenChangelogIsNotForcedButChangelogSubjectExists()
+      throws IOException, RestClientException
+  {
+    // Given:
+    final String sourceSubject = TOPIC_NAME + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX;
+    final String changelogTopic = "_confluent-ksql-default_query_ID-base-Reduce-changelog";
+    when(queryBuilder.getKsqlConfig())
+        .thenReturn(KSQL_CONFIG.cloneWithPropertyOverwrite(
+            ImmutableMap.of(KsqlConfig.SCHEMA_REGISTRY_URL_PROPERTY, "localhost:1234")));
+    when(queryBuilder.getQueryId()).thenReturn(new QueryId("ID"));
+    when(srClient.getAllSubjects()).thenReturn(ImmutableSet.of(sourceSubject));
+    when(topicClient.isTopicExists(changelogTopic)).thenReturn(true);
+    givenUnwindowedSourceTable(false);
+
+    // When:
+    tableSource.build(planBuilder);
+
+    // Then:
+    verify(srClient, Mockito.never()).register(anyString(), any(ParsedSchema.class));
+  }
```

Comment:
Add a test to check that we don't try to register schemas if schema registry is not enabled?